hamlpy/parser/filters.py

```diff
@@ -37,6 +37,10 @@ def plain(content, indent, options):
     return textwrap.dedent(content)
 
 
+def preserve(content, intent, options):
+    return textwrap.dedent(content.replace('\n', '&#10;').replace('\r', '&#13;'))
```

Comment:
This should match https://github.com/haml/haml/blob/master/lib/haml/helpers.rb#L132